src/libs/actions/BankAccounts.js

```diff
@@ -270,12 +272,352 @@ function fetchUserWallet() {
+
+            // If we're not in setup, it means we already have a withdrawal account and we're upgrading it to a business
+            // bank account. So let the user review all steps with all info prefilled and editable, unless a specific
+            // step was passed.
+            if (!achData.isInSetup) {
+                // @TODO Not sure if we need to do this since for NewDot none of the accounts are pre-existing ones
+                currentStep = '';
+            }
+
+            // Temporary fix for Onfido flow. Can be removed by nkuoch after Sept 1 2020.
+            // @TODO not sure if we still need this or what this is about, but seems like maybe yes...
+            if (currentStep === CONST.BANK_ACCOUNT.STEP.ACH_CONTRACT && achData.useOnfido) {
+                const onfidoRes = lodashGet(achData, CONST.BANK_ACCOUNT.VERIFICATIONS.REQUESTOR_IDENTITY_ONFIDO);
+                const sdkToken = lodashGet(onfidoRes, CONST.BANK_ACCOUNT.ONFIDO_RESPONSE.SDK_TOKEN);
+                if (sdkToken && !achData.isOnfidoSetupComplete
+                        && onfidoRes.status !== CONST.BANK_ACCOUNT.ONFIDO_RESPONSE.PASS
+                ) {
+                    currentStep = CONST.BANK_ACCOUNT.STEP.REQUESTOR;
+                }
+            }
+
+            // Ensure we route the user to the correct step based on the status of their bank account
+            if (bankAccount && !currentStep) {
+                currentStep = bankAccount.isPending() || bankAccount.isVerifying()
+                    ? CONST.BANK_ACCOUNT.STEP.VALIDATION
+                    : CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+
+                // @TODO Again, not sure how much of this logic is needed right now as we shouldn't be handling any
+                // open accounts in E.cash yet that need to pass any more checks or can be upgraded, but leaving in for
+                // possible future compatibility.
+                if (bankAccount.isOpen()) {
+                    if (bankAccount.needsToPassLatestChecks()) {
+                        const hasTriedToUpgrade = bankAccount.getDateSigned()
+                            > (kycVerificationsMigration || '2020-01-13');
+                        currentStep = hasTriedToUpgrade
+                            ? CONST.BANK_ACCOUNT.STEP.VALIDATION : CONST.BANK_ACCOUNT.STEP.COMPANY;
+                        achData.bankAccountInReview = hasTriedToUpgrade;
+                    } else {
+                        // We're not going to have a EnableStep, but we can show this account as open if accessed
+                        currentStep = CONST.BANK_ACCOUNT.STEP.ENABLE;
+                    }
+                }
+            }
+
+            // If at this point we still don't have a current step, default to the BankAccountStep
+            if (!currentStep) {
+                currentStep = CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+            }
+
+            Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {throttledDate});
```

Comment:
Lmk if that helps think about what is happening in a different way or if you still think I'm missing something. 

>in any event it seems like a bad pattern / unnecessary to have previousACHData be connected to Onyx

Definitely open to alternative suggestions here.

Reply:
Sorry, I was confused here. I was thinking that `throttledDate` was essentially ACH data, not an actual **date**, and that this would overwrite the `previousACHData` with the data inside this `throttledData` object. So yeah this seems fine. Sorry for the confusion.